staging/src/k8s.io/legacy-cloud-providers/azure/azure_instance_metadata.go

```diff
@@ -145,7 +145,7 @@ func (ims *InstanceMetadataService) getInstanceMetadata(key string) (interface{}
 
 // GetMetadata gets instance metadata from cache.
 func (ims *InstanceMetadataService) GetMetadata() (*InstanceMetadata, error) {
-	cache, err := ims.imsCache.Get(metadataCacheKey)
+	cache, err := ims.imsCache.Get(metadataCacheKey, cachedData)
```

Comment:
Can we reverse trace this call? I don't think instance metadata change at all. and it should be safe to do unsafe reads here.

Reply:
[Node Addresses](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go#L44) is the one which requests the instance metadata and [GetZone](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/azure/azure_zones.go#L54) are the only 2 calls for `GetMetadata`. 

I think `GetZone` should be safe with stalled read.